network/network.go

```diff
@@ -76,8 +75,9 @@ type Network struct {
 	decrypter              crypto.Decrypter
 	nodeDIDResolver        transport.NodeDIDResolver
 	didDocumentFinder      types.DocFinder
-	connectionsDB          *bbolt.DB
+	connectionStore        storage.KVStore
 	eventPublisher         events.Event
+	warehouse              storage.Warehouse
```

Comment:
warehouse is a weird term, never encountered in IT before

Reply:
Data Warehouse (not exactly the same though).

I thought: what holds/provides stores? -> Warehouses (to avoid having 1000 `*Manager` types)

Reply:
And a Factory doesn't creates stores. Builders do, but they are already reserved for a different kind of configuring....
`Provider` would be doable? `storage.Provider` would be the type then

Reply:
I'll change it to `Provider`.